ocs_ci/utility/aws.py

```diff
@@ -262,50 +269,75 @@ def detach_and_delete_volume(self, volume, timeout=120):
-            ):
-                if sample == STOPPED:
-                    return True
-            return False
-        state = res.get('StoppingInstances')[0].get('CurrentState').get('Code')
-        return state == STOPPING
-
-    def start_ec2_instance(self, instance_id, wait=False):
+            for instance_id, instance_name in instances.items():
+                logger.info(
+                    f"Waiting for instance {instance_name} to reach status stopped"
+                )
+                instance = self.get_ec2_instance(instance_id)
+                instance.wait_until_stopped()
+
+    def start_ec2_instances(self, instances, wait=False):
         """
         Starting an instance
 
         Args:
-            instance_id (str): ID of the instance to start
+            instances (dict): A dictionary of instance IDs and names to start
             wait (bool): True in case wait for status is needed,
                 False otherwise
 
-        Returns:
-            bool: True in case operation succeeded, False otherwise
         """
-        res = self.ec2_client.start_instances(InstanceIds=[instance_id])
-        if wait:
-            for sample in utils.TimeoutSampler(
-                TIMEOUT, SLEEP, self.get_instances_status_by_id, instance_id
-            ):
-                if sample == RUNNING:
-                    return True
-            return False
-        state = res.get('StartingInstances')[0].get('CurrentState').get('Code')
-        return state == PENDING
+        instances = {
+            key: val for key, val in instances.items() if (
+                self.get_instances_status_by_id(key) == constants.INSTANCE_STOPPED
+            )
+        }
+        if instances:
+            instance_ids = [key for key in instances.keys()]
+            instance_names = [val for val in instances.values()]
+            logger.info(f"Starting instances {instance_names}")
+            assert self.ec2_client.start_instances(InstanceIds=instance_ids), (
+                f"Failed to start ec2 instances {instance_names}"
+            )
+            if wait:
+                for instance_id, instance_name in instances.items():
+                    logger.info(
+                        f"Waiting for instance {instance_name} to reach status running"
+                    )
+                    instance = self.get_ec2_instance(instance_id)
+                    instance.wait_until_running()
+
+
+def get_instances_ids_and_names(instances):
+    """
+    Get the instances IDs and names according to nodes dictionary
+
+    Args:
+        instances (list): Nodes dictionaries, returned by 'oc get node -o yaml'
+
+    Returns:
+        dict: The ID keys and the name values of the instances
+
+    """
+    return {
+        'i-' + instance.get('spec').get('providerID').partition('i-')[
+            -1
+        ]: instance.get('metadata').get('name') for instance in instances
```

Comment:
The wrapping here makes this highly unreadable

Reply:
done